### ISSUE TITLE
How does type extension work?

### ISSUE THREAD

**Issue by tonyxiao** [graphql/graphql-js#483]
I see some [commits](https://github.com/graphql/graphql-js/pull/174/files) that adds the following syntax

```
type Hello {
  world: String
}
extend type Hello {
  world2: String
}
```

However I can't find any docs around just what it is and how to use it. And if I actually try to use it in code I get the following message.

```
Error: Client Schema cannot be used for execution.
```

Can someone clarify how to use type extension in GraphQL?

**Comment by robzhu** [comment]
Hey @tonyxiao, can you please post the code you're running to execute type extension?

**Comment by tonyxiao** [comment]
Hmm I was using it together with ApolloStack. apollostack.com Maybe they are using an older version of graphql js and that's why there's error?

**Comment by stubailo** [comment]
Sounds like it should be an issue on graphql-tools? https://github.com/apollostack/graphql-tools

**Comment by robzhu** [comment]
Closing this since I haven't seen any movement on this for a while. @tonyxiao, please feel free to re-open if feel there's something specific and actionable we can do in this repo.

**Comment by stubailo** [comment]
feel free to open an issue on graphql tools about documenting this.

**Comment by dyst5422** [comment]
I'm still seeing `extend` not work at all with graphql or express-graphql. Minimal failing example attached.

```ts
import { buildSchema, graphql } from 'graphql';

const schema = buildSchema(`
type QuerySubset {
  hello: String
}

type Queries {
  querySubset: QuerySubset
}

extend type Queries {
  hello: String
}

type Query {
  Queries: Queries
}

schema {
  query: Query
}
`)

const rootResolver = {
  Queries: {
    querySubset: {
      hello: () => 'hello world',
    },
    hello: () => 'hello world',
  }
};

// Works fine
graphql(schema, `{ Queries { querySubset { hello } } }`, rootResolver).then(response => {
  console.log(response);
})

// Errors out with 'Cannot query field "hello" on type "Queries".'
graphql(schema, `{ Queries { hello } }`, rootResolver).then(response => {
  console.log(response);
})
```

**Comment by krabbypattified** [comment]
Why is this closed? I would like to know how the `extend` keyword works and it's not documented.

**Comment by dyst5422** [comment]
@krabbypattified See the closed issue #892 and associated issue #922

**Comment by dandv** [comment]
TL;DR - [use graphql-tools for now](https://stackoverflow.com/questions/48917863/in-graphql-how-do-you-do-basic-extend-types/56204966#56204966).
